README.md

```diff
@@ -109,135 +49,220 @@ You should use the following process as much as possible:
+
+### Tests
+
+Start by running the tests:
 
-### Baseline
+```
+$ rake test
+... test output ...
+Finished in 0.14685s
+28 tests, 64 assertions, 0 failures, 0 errors, 1 skips
+```
 
-To start this project, take some time to get familiar with the code. Do the following in this order:
-1. Read through all of the tests
-1. Look at the provided CSV files: `support/drivers.csv`, `support/users.csv`, `support/trips.csv`
-1. Then look through the ruby files in the `lib` folder
+All the existing code is thoroughly tested. Some of the code you write for this project will break these tests; when that happens, it is your job to update them. When you add new functionality to this project, you should add new tests as well.
 
-Create a diagram that describes how each of these classes and methods (messages) interact with one another as well as with the CSV files.
+The tests can serve as an example of how the methods and classes should work. If you're ever confused about what one of these methods looks like "from the outside", the tests can serve as your guide.
 
-**Exercise:** Look at this requirement in Wave 1: "For a given user, calculate their total expenditure for all trips". Spend some time writing pseudocode for this.
+### Reading Code
 
-### Wave 1
+The existing code contains 4 classes:
+- `Passenger`
+- `Trip`
+- `CsvRecord`
+- `TripDispatcher`
+
+**Before you go any further, think about how these classes might be related.** Draw a diagram, if that will help. Your guess doesn't have to be correct at this stage - making a prediction and then checking it is an important part of the learning process.
+
+Now, start reading through the code. There are two equally valid ways to approach this:
+
+- Top-down: start with the big pieces.
+    - What classes are there, and how do they interact?
+    - How does it look from the outside?
+    - How are these classes tested?
+- Bottom-up: start with the details. Pick an interesting method (say `Passenger.load_all`) and learn everything about it.
+    - What arguments does it take?
+    - How does it work? If there's more than one step, what are they? Why are they in that order?
+    - What other methods does call? 
+    - What does it return?
+    - What code calls it, and how does it use the returned value?
+    - How is it tested?
+
+### Playing Around
+
+One of the most effective way to learn a new codebase is to use it. Since this code is a library not a program, there's nothing obvious to run from the command line, so we'll have to be a little clever. We can use `pry` to load the code, and then play with it in the interpreter:
+
+```
+$ pry -r ./lib/trip_dispatcher.rb
+[1] pry(main)> td = RideShare::TripDispatcher.new
+=> #<RideShare::TripDispatcher:0x3fe91f52950c>
+[2] pry(main)> td.trips.length
+=> 600
+```
```

Comment:
These sections are great! I like the distinction between top-down and bottom-up approaches